lib/rubocop/cop/lint/ambiguous_block_association.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for ambiguous block association with method
+      # when param passed without parentheses.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # It's ambiguous because there is no parentheses around `a` param
+      #   some_method a { |val| puts val }
+      #
+      # @example
+      #
+      #   # good
+      #
+      #   # With parentheses, there's no ambiguity.
+      #   some_method(a) { |val| puts val }
+      class AmbiguousBlockAssociation < Cop
+        MSG = 'Parenthesize the param `%s` to make sure that block will be '\
+              'associated with `%s` method call.'.freeze
+
+        def on_send(node)
+          return if node.parenthesized? || node.assignment? || node.method?(:[])
+
+          return unless method_with_block?(node.first_argument)
+          first_child = node.first_argument.children.first
```

Comment:
I'm going to rename `first_child` to `first_param` and stick to `param` naming instead of `args`. Just to be consistent.